frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,54 +206,238 @@ export function instructorsPermissionTests(api) {
+            "instructor_permission_instructor_update_test@test.com"
+        );
+    });
 
     it("fetch sessions", async () => {
-        let resp = await apiGET("/instructor/instructors");
+        await switchToInstructorOnlyUser();
+        let resp = await apiGET("/instructor/sessions");
+        expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET("/instructor/sessions");
         expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update session");
+    it("can't update session", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newSession = {
+            start_date: "2020-04-10T01:00:00.000Z",
+            end_date: "2020-11-31T00:01:20.000Z",
+            name: "instructor_permission_cant_update_session_test_session",
+        };
+
+        let resp = await apiPOST("/instructor/sessions", newSession);
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST("/admin/sessions", newSession);
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
 
     it("fetch positions", async () => {
+        await switchToInstructorOnlyUser();
         let resp = await apiGET(`/instructor/sessions/${session.id}/positions`);
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET(`/instructor/sessions/${session.id}/positions`);
+        expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update position");
+    it("can't update position", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newPosition = {
+            position_code:
+                "instructor_permission_cant_update_position_test_code",
+            position_title:
+                "instructor permission cant update position test position",
+            hours_per_assignment: 1.0,
+            contract_template_id: existingContractTemplateId,
+            duties: "Tutorials",
+        };
+
+        let resp = await apiPOST(
+            `/instructor/sessions/${session.id}/positions`,
+            newPosition
+        );
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/positions`,
+            newPosition
+        );
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(`/admin/positions`, newPosition);
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(`/instructor/positions`, newPosition);
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
 
     it("fetch contract templates", async () => {
+        await switchToInstructorOnlyUser();
         let resp = await apiGET(
             `/instructor/sessions/${session.id}/contract_templates`
         );
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
```

Comment:
Why? You already tested fetching as the instructor....

Reply:
For the first test, the user has only the instructor role assigned; for the second test, it is a user with mixed roles (admin, ta and instructor). The intuition here is to test with a different combination of permissions and assure that as long as the instructor is one of the roles, this route would be accessible. (it may be redundent)